src/checkpointTrie.ts

```diff
@@ -147,10 +150,37 @@ module.exports = class CheckpointTrie extends BaseTrie {
    * @method createScratchReadStream
    * @private
    */
-  _createScratchReadStream (scratch) {
+  _createScratchReadStream(scratch: ScratchDB) {
     scratch = scratch || this._scratch
     const trie = new BaseTrie(scratch._leveldb, this.root)
     trie.db = scratch
     return new ScratchReadStream(trie)
   }
+
+  // formats node to be saved by levelup.batch.
+  // returns either the hash that will be used key or the rawNode
+  _formatNode(node: TrieNode, topLevel: boolean, opStack: BatchDBOp[], remove: boolean = false) {
+    const rlpNode = node.serialize()
+
+    if (rlpNode.length >= 32 || topLevel) {
+      const hashRoot = node.hash()
+
+      if (remove && this.isCheckpoint) {
+        opStack.push({
+          type: 'del',
+          key: hashRoot,
+        })
+      } else {
+        opStack.push({
+          type: 'put',
+          key: hashRoot,
+          value: rlpNode,
+        })
+      }
+
+      return hashRoot
+    }
+
+    return node.raw
+  }
```

Comment:
Ah, just seeing this which is likely related to my `_formatNode` question in `baseTrie`, could you nevertheless give a short explanation of the effects here? Does this fixes a bug? Make things more secure?